folktables/utils/files_resources.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC
+from dataclasses import dataclass
+import pathlib
+
+
+@dataclass
+class FilesResource(ABC):
```

Comment:
Why is this an ABC?

Reply:
My initial intention to make `FilesResource` an `ABC` was to force developers to make child classes for each `DataSource` type. For example, the `ACSDataSource` would have its own `ACSFilesResource`. Likewise, SIPP would have `SIPPFilesResource`. This would ensure that the `FilesResource`s for each `DataSource` had the same "structure" while also allowing future developers to extend the class to better fit their needs.

Now that you point it out, it might be better to allow this to be a concrete class (which would also address some of your comments below) which would avoid making developers create an empty class that inherits from `FilesResource` in case they don't want to extend it.

Reply:
OK thanks for explaining the reasoning! As long as all the derived classes will usually have the same fields and we're not doing explicit type checking with the derived classes, it's probably good to keep things simple and just use `FilesResource` as a regular dataclass.